rxandroidble/src/main/java/com/polidea/rxandroidble/RxBleConnection.java

```diff
@@ -417,4 +423,23 @@ public String toString() {
      */
     @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
     Observable<Integer> requestMtu(int mtu);
+
+    /**
+     * <b>This method requires deep knowledge of RxAndroidBLE internals. Use it only as a last resort if you know
+     * what your are doing.</b>
+     * <p>
+     * Queue an operation for future execution. The method accepts a builder parameter. The builder function
+     * will be invoked with three arguments and should return the a non-null instance of {@link RxBleRadioOperation}.
+     * The three arguments the builder receives are:
+     * <ol>
+     * <li>{@link Scheduler} The RxBleRadio scheduler used to execute operation ({@link AndroidSchedulers#mainThread()}</li>
+     * <li>{@link BluetoothGatt} The Android API GATT instance</li>
+     * <li>{@link RxBleGattCallback} The internal Rx ready bluetooth gatt callback to be notified of GATT operations</li>
+     * </ol>
+     *
+     * @param builder The builder used to create the {@link RxBleRadioOperation} instance
+     * @param <T>     The type returned by the {@link RxBleRadioOperation} instance
+     * @return Observable emitting the value after execution or an error in case of failure.
+     */
+    <T> Observable<T> queue(Func3<Scheduler, BluetoothGatt, RxBleGattCallback, RxBleRadioOperation<T>> builder);
```

Comment:
I'm all in for 1 and 2, I thought about passing them, but wasn't sure, will change.

For 3, do you see `RxBleRadioOperationCustom` as `RxBleRadioOperationCustom extends RxBleRadioOperation` or as a standalone class (would probably be an interface to me).

I see it as this (I would like to come up with a better name than `RxBleRadioOperationCustom` however):

```
    interface RxBleRadioOperationCustom<T> {
        void execute(BluetoothGatt bluetoothGatt,
                            RxBleGattCallback rxBleGattCallback,
                            Scheduler scheduler,
                            Observer<T> observerProxy,
                            Action0 releaseRadioAction) {
        } throws Exception
    }
```

Then the `queue` would adapt this interface to an `RxBleRadioOperation`:

```
    @Override
    public <T> Observable<T> queue(final RxBleRadioOperationCustom<T> operation) {
        return rxBleRadio.queue(new RxBleRadioOperation<T>() {
            @Override
            protected void protectedRun() throws Throwable {
                operation.execute(bluetoothGatt, gattCallback, rxBleRadio.scheduler(), asObserver(), new Action0() {
                    @Override
                    public void call() {
                        releaseRadio();
                    }
                });
            }

            @Override
            protected BleException provideException(DeadObjectException deadObjectException) {
                return new BleDisconnectedException(deadObjectException, bluetoothGatt.getDevice().getAddress());
            }
        });
    }
```

The gain I see here is that `RxBleRadioOperation` is not exposed to consumer of the library meaning you are still free to refactor it as you see fit. You only need to keep backward-compatibility on the `RxBleRadioOperationCustom` interface now.

What do you think?